server/src/repositories/Training/ModuleRepository.ts

```diff
@@ -24,8 +24,9 @@ export async function getModules(): Promise<TrainingModule[]> {
   return TrainingModuleMap.trainingModulesToDomain(knexResult);
 }
 
-export async function deleteModuleById(id: number): Promise<void> {
-  await knex("TrainingModule").where({ id: id }).del();
+export async function archiveModuleById(id: number): Promise<TrainingModule | null> {
```

Comment:
I think we can rename this to just `archiveModule`, since "by ID" is implied by the argument. This will also make it consistent with `archiveModuleItem`.